### ISSUE TITLE
Switching to next wallpaper deactivates all shell extensions

### ISSUE THREAD

**Issue by paddyesch** [rodakorn/randwall#21]
I used the 'next' button to switch to the next wallpaper and all shell extensions were deactivated. However, it doesn't happen every time I click 'next', only sometimes. I'm using Fedora24 (GNOME Shell 3.20.3) and the latest version of randwall from extensions.gnome.org.

Otherwise, I'm very happy with this extension, thank you!

**Comment by rodakorn** [comment]
Hi!

Mmmm... I cannot reproduce that behaviour on a fedora 24 with gnome-shell 3.20.3. I know it's difficult but if you could add some logs when that behaviour occurs I will try to check what happens.

**Comment by redianthus** [comment]
I can confirm this: I used to have the same issue. But now, as I let the extension running in the background and don't touch it, I don't experience it anymore. It was on Debian Unstable, probably not with 3.20 at the time, but I'm not sure...

**Comment by mrEDitor** [comment]
I have this issue too, and the only known to me way to reproduce it ­— change wallpapers _really_ lot of times. But sometimes it appears when background is changing by timer, too.

I don't have any warnings or errors when it happens, just reboot logs of Gnome Shell and extensions.
Also it looks like that if I just remove `Gio.Settings.set_string(...)` code, all is working well (except wallpapers do not change, of course), so I think it might be Gnome Shell bug or some kind of race condition.

**Comment by rodakorn** [comment]
Ok. For me it's a little difficult to test it with a fedora because I use debian... actually yesterday I asked a friend of mine that uses fedora to check that issue but he couldn't reproduce it. But seeing that some of you confirm this issue and is not an isolated problem I will make a VM to properly test it.

Thanks everyone for the feedback.

**Comment by mrEDitor** [comment]
Actually, I use Archlinux.

What is about the bug, something like

```
 settings = new Gio.Settings({schema: 'org.gnome.desktop.background'})
 picture = settings.get_string('picture-uri')
 for (let i = 0; i < 150; i++) settings.set_string('picture-uri', picture)
```

in Looking Glass will give same result (but will make more damage to Shell instance 'cause runs under lg and forces nearly whole shell to restart).

**Comment by redianthus** [comment]
@rodakorn I was/am on Debian too. IMO it's not specific to Fedora.

**Comment by rodakorn** [comment]
I don't think it's fedora specific either, but on a debian with gnome-shell 3.14.4 I cannot reproduce this issue (later I will try mrEDitor suggestion) but I thought maybe it has something to do with gnome-shell version because is not the first issue that I receive from people using gnome-shell 3.20... So I'll try with a Fedora that uses a 3.20 gnome-shell version

**Comment by redianthus** [comment]
At the time I was not on 3.14 but 3.16 or 3.18 I'm not sure sorry.

**Comment by paddyesch** [comment]
Correction: I can reproduce this bug. It just happens really rarely by clicking on the 'next' button.

I recorded the output of this command while the shell crashed:

`journalctl /usr/bin/gnome-shell -f -o cat`

http://pastebin.com/hc1vjGZg

The crash occurred after line 10 I think. Hope it helps.

~~I'm afraid I can't reproduce this issue either. I didn't really change anything (other than playing around with unrelated extensions) but continuously switching to the next wallpaper works fine now.~~

~~A wild guess: I experienced this bug right after installation several times but maybe a reboot fixed it somehow.~~

**Comment by urandom** [comment]
On arch linux (gnome 3.20), I've been having this issue for quite a while now. It's not related to the manual wallpaper switch, but changing the wallpaper itself. I've been noticing that after a while, the extension will crash and take all extensions after it (alpha) with it. I only managed to pinpoint the problem to the actual switching when I caught it crashing while the wallpaper was being switched automatically at some point, and, iirc, the laptop froze for a second or so (I guess a lot of i/o was happening or something).

**Comment by redianthus** [comment]
Indeed, with a little time value for changing the wallpaper it happens again. I though it was when switching the wallpaper manually but it's more like @urandom just said.

So, in a terminal:

`gnome-shell --replace &`

Then, I switched the wallpaper a lot of time and when the bug appeared, I had this [output](https://gist.github.com/zapashcanon/4ce5bdbe91122ec92fc6713b84a4939e) on my terminal.

Don't know if it helps. @rodakorn

**Comment by danielquinn** [comment]
Perhaps it's not the plugin at all, but the image randomly selected?  Has anyone managed to intentionally reproduce this problem in a reliable way?

**Comment by redianthus** [comment]
@danielquinn: I tried another extension (BackSlide) with the same set of images, I had no problem. The main differences with randwall is that rand wall search for images in a given folder when in BackSlide you have to add images to the list yourself (ie: it won't search for new images in a folder). So I think it comes from randwall. Moreover, you can reproduce this problem just like I said: switching to the next wallpaper many times and it will happen. (That said, I didn't try since my last comment)

**Comment by simonbcn** [comment]
Similar problem on Arch Linux 64 bits and GNOME Shell 3.22.2.
In my case, sometimes, when I _waked up_ the session after screen goes off, the extensions weren't enabled again but "gsettings get org.gnome.shell enabled-extensions" showed the right value. To "solve it" I had to relogin to gnome session. https://bugs.archlinux.org/task/52490
After disabling this extension, the problem gone.

**Comment by j-keiter** [comment]
This problem has persisted for last year on 3 different computers from ubuntu gnome 15.04 to 16.04.  Love the extension but It always randomly shuts off all other extensions about every 10min to an hour after restarting gnome-shell.  I attached a copy of my journalctl just after all my extension suddenly turned off.   
[journalctl_output.txt](https://github.com/rodakorn/randwall/files/751095/journalctl_output.txt)  Wish I could help fix.

**Comment by mrEDitor** [comment]
Just a feedback to extension developer and users: noticed no crashes anymore for a long time. I am on Archlinux with Gnome Shell 3.22.3-2 / Gnome Desktop 3.22.2-1 currently. 
Any more remarks?

**Comment by simonbcn** [comment]
I'll test but the last reports: https://github.com/rodakorn/randwall/issues/21#issuecomment-273427815 and https://github.com/rodakorn/randwall/issues/21#issuecomment-277303925 are subsequent to the last significant change in the code: https://github.com/rodakorn/randwall/commit/ff4715a5ed78ad5b4bfa4ca6d0204edfa71b19ee

**Comment by mrEDitor** [comment]
I'm still pretty sure that the bug was in the Gnome Shell itself, not in the extension.

**Comment by simonbcn** [comment]
@mrEDitor you are wrong. This extension is really buggy. I have installed yesterday and now I can reproduce this issue. The problem now is worst for me:
1. The desktop hangs several seconds.
2. All extensions are disabled.

**Comment by ieugen** [comment]
It deactivates all extensions for me too. Things started happening like this after I installed it. 
Once I deactivated it, everithing is ok. 

Using Debian Stretch with Gnome 3.22 . 

It is really buggy.

**Comment by Vaesper** [comment]
Arch Linux 64 bit with Gnome 3.24 on Wayland... Crashes the shell on both my laptop and desktop (similar setups). More often on desktop where my wallpaper resolutions are bigger.
